hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -601,6 +605,16 @@ public OzoneOutputStream createKey(
     HddsClientUtils.verifyResourceName(volumeName, bucketName);
     HddsClientUtils.checkNotNull(keyName, type, factor);
     String requestId = UUID.randomUUID().toString();
+
+    if(Boolean.valueOf(metadata.get(OzoneConsts.GDPR_FLAG))){
```

Comment:
@ajayydv Yes, it can be done; but then the cost of this function will be borne by the data node. Today the EC of Hadoop/Ozone pushes that to the client; so the data nodes don't see this cost. Also, today we have no mechanism to deliver the key to data node; that means even if we lose the HDD disk, the person who finds it cannot decode it; since data node never sees the key. 

> From security side that will be more secure as we will not be sharing it over the wire on client side.

Completely agree; we will need to make sure that RPC is secure on wire when we send this key. But remember; GDPR is not about security; it is about forgetting the key when we delete a block; so that we can write and sign a document saying that file has been deleted. Since it is not a security feature, even if the key is leaked; it is ok; all we are saying -- or promising is that via Ozone Manager you cannot get to the key. if someone has made a copy of that file; it is a problem, but something that we do not know off.

@dineshchitlangia  has attached a design document here, where he discusses some of these issues. https://issues.apache.org/jira/browse/HDDS-2012